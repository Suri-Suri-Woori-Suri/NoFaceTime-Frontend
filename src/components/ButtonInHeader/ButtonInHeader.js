import React from 'react';
import styles from './ButtonInHeader.module.css';

const ButtonInHeader = ({
  isLoggedIn,
  updateUserData
}) => {
  const buttonClickHandler = () => {
    if (!isLoggedIn) return;

    updateUserData({ isLoggedIn: false });
  };

  return (
    <div className={styles.ButtonInHeader}>
      <button
        className={styles.Button}
        onClick={buttonClickHandler}>
        {isLoggedIn ? 'Sign Out' : 'Sign In'}
      </button>
    </div>
  );
};

export default ButtonInHeader;
