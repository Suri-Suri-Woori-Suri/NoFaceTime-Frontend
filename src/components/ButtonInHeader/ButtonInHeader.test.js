import React from 'react';
import { mount } from 'enzyme';

import ButtonInHeader from './ButtonInHeader';

describe('<ButtonInHeader />', () => {
  const mockFn = jest.fn();

  const wrapperLogin = mount(
    <ButtonInHeader
      isLoggedIn={false}
      updateUserData={mockFn} />
  );

  const wrapperLogout = mount(
    <ButtonInHeader
      isLoggedIn={true}
      updateUserData={mockFn} />
  );

  it('should be contain login button', () => {
    const loginButton = wrapperLogin.findWhere(node =>
      node.type() === 'button');

    expect(loginButton.text()).toBe('Sign In');
  });

  it('should be contain logout button', () => {
    const logoutButton = wrapperLogout.findWhere(node =>
      node.type() === 'button');

    expect(logoutButton.text()).toBe('Sign Out');
  });


  it('matches snapshot', () => {
    expect(wrapperLogin).toMatchSnapshot();
    expect(wrapperLogout).toMatchSnapshot();
  });
});
