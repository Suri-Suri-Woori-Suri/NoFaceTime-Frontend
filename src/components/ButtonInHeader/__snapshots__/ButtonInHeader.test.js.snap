// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonInHeader /> matches snapshot 1`] = `
<ButtonInHeader
  isLoggedIn={false}
  updateUserData={[MockFunction]}
>
  <div
    className="ButtonInHeader"
  >
    <button
      className="Button"
      onClick={[Function]}
    >
      Sign In
    </button>
  </div>
</ButtonInHeader>
`;

exports[`<ButtonInHeader /> matches snapshot 2`] = `
<ButtonInHeader
  isLoggedIn={true}
  updateUserData={[MockFunction]}
>
  <div
    className="ButtonInHeader"
  >
    <button
      className="Button"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </div>
</ButtonInHeader>
`;
